### PR TITLE
Support constant globals in precompute pass

### PR DESCRIPTION
**Pull request by dcodeIO**
This PR includes non-mutable globals in precompute, which will allow me to continue removing manual inlining of constants in AssemblyScript without breaking something. Related: https://github.com/WebAssembly/binaryen/issues/1621, i.e.

```ts
enum Animal {
  CAT = 0,
  DOG = CAT + 1 // requires that `Animal.CAT` is evaluated to
                // precompute the constant value for `Animal.DOG`
}
```

### CODE REVIEW

**Comment by dcodeIO** [comment]
(going to fix the tests tomorrow, if it's otherwise good)

File: src/passes/Precompute.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ typedef std::unordered_map<GetLocal*, Literal> GetValues;
 
 // Execute an expression by itself. Errors if we hit anything we need anything not in the expression itself standalone.
 class StandaloneExpressionRunner : public ExpressionRunner<StandaloneExpressionRunner> {
```
Comment (kripken) [inline src/passes/Precompute.cpp]: The name of the class, and comment above, are outdated with this change. How about `PrecomputingExpressionRunner`?

File: src/passes/Precompute.cpp [hunk 2]
```diff
@@ -87,6 +89,12 @@ class StandaloneExpressionRunner : public ExpressionRunner<StandaloneExpressionR
     return Flow(NONSTANDALONE_FLOW);
   }
   Flow visitGetGlobal(GetGlobal *curr) {
+    auto global = module->getGlobalOrNull(curr->name);
```
Comment (kripken) [inline src/passes/Precompute.cpp]: `auto*` (with `*`, that ensures it's a pointer type, in the unlikely chance the function is later changed to return something else)

File: test/merge/basics.wast.combined.finalized.opt [hunk 3]
```diff
@@ -7,12 +7,8 @@
  (import "env" "some-collide" (func $some-collide))
  (import "env" "some-func-b" (func $some-func-b))
  (import "env" "some-collide" (func $some-collide$0))
- (global $global-collide i32 (i32.const 0))
```
Comment (kripken) [inline test/merge/basics.wast.combined.finalized.opt]: Hmm, this optimization is too good, it makes some testcases less interesting ;)

How about changing these merge testcase globals to be mutable, so they aren't optimized out?

Comment (dcodeIO) [reply test/merge/basics.wast.combined.finalized.opt]: Ah, right, made it so that there's a mutable and a constant global for each one.

**Comment by kripken** [comment]
Thanks @dcodeIO!

**Pull request merged**